pkg/resource/managed_reconciler.go

```diff
@@ -406,6 +410,23 @@ func (r *ManagedReconciler) Reconcile(req reconcile.Request) (reconcile.Result,
 			return reconcile.Result{RequeueAfter: r.shortWait}, errors.Wrap(r.client.Status().Update(ctx, managed), errUpdateManagedStatus)
 		}
 
+		if !managed.IsConditionTrue(v1alpha1.TypeReferencesResolved) {
+			if err := r.managed.ResolveReferences(ctx, managed); err != nil {
+				// update the status according to the type of the err
+				switch err.(type) {
+				case *resourceNotReadyErr:
+					managed.SetConditions(v1alpha1.ResolveReferencesBlocked(err))
+				default:
+					managed.SetConditions(v1alpha1.ResolveReferencesError(err))
```

Comment:
On reading this block I'd prefer to avoid introducing a distinct `ResolveReferencesError`.

Every other case in which we encounter an unexpected error during reconciliation is reflected using `ReconcileError`, making this the only case we do something different.